service-python/pdfgenerator/src/lib/queues.py

```diff
@@ -21,17 +20,21 @@ def on_generate_callback(channel, method, properties, body):
 
         # binding_key = method.routing_key
         message = json.loads(body)
-        # logging.info(f" [x] {binding_key}: Received message: {message}")
+        # logging.info(f" [x] Received message: {message}")
         claim_id = message["claimSubmissionId"]
+        diagnosis_code = message["diagnosticCode"]
         message["veteran_info"] = message["veteranInfo"]
-        message["evidence"] = message["evidence"]
-        code = message["diagnosticCode"]
-        diagnosis_name = DIAGNOSTIC_CODE_MAPPING[code]
-        variables = pdf_generator.generate_template_variables(diagnosis_name, message)
+        if message['pdfTemplate'] in ['v1', 'v2']:
+            pdf_template = message['pdfTemplate']
+        else:
+            # Default to version 1
+            pdf_template = "v1"
+        template_name = codes[diagnosis_code] + "-" + pdf_template
+        variables = pdf_generator.generate_template_variables(template_name, message)
         # logging.info(f"Variables: {variables}")
-        template = pdf_generator.generate_template_file(diagnosis_name, variables)
-        pdf = pdf_generator.generate_pdf_from_string(diagnosis_name, template, variables)
-        redis_client.save_hash_data(f"{claim_id}-pdf", mapping={"contents": base64.b64encode(pdf).decode("ascii"), "diagnosis": diagnosis_name})
+        template = pdf_generator.generate_template_file(template_name, variables)
+        pdf = pdf_generator.generate_pdf_from_string(template_name, template, variables)
+        redis_client.save_hash_data(f"{claim_id}-pdf", mapping={"contents": base64.b64encode(pdf).decode("ascii"), "diagnosis": codes[diagnosis_code]})
```

Comment:
so we switched to diagnosis code for file names?

Reply:
Essentially yes, it saves us from needing any non-intuitive/non-standard mappings.

Reply:
Functionality does not change for the filename. It will still be in the same format as `VAMC_Hypertension_Rapid_Decision_Evidence--20230103%20(4).pdf` this is just cleaning up the variables in the code. `codes[diagnosis_code]` will fetch the diagnosis name from the `settings` file

Reply:
Oh @anthonyvelazquez is right, I misread the question